samples/kubernetes/graceful-shutdown/graceful-shutdown.md

```diff
@@ -53,50 +53,66 @@ app.Use((httpContext, next) =>
 
 Services that handle requests coming from [Kubernetes ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) may run into an issue where they continue to receive new requests *after* Kubernetes requests them to shut down via `SIGTERM` signal. This usually lasts for only a few seconds, but if the service shuts down immediately and these requests fail, it puts a burden on the client to retry them and the overall service responsiveness goes down.
 
-The reason for this behavior is that ingress and Kubernetes control plane are separate entities, operating independently. When Kubernetes decides to shut down some pod(s), there is inevitable delay between control plane taking a pod out of Service backend, and ingress "noticing" that this has happened and actually stopping routing requests to that pod. So it is useful to introduce some delay between the time the pod receivers `SIGTERM` signal, and the time when it stops responding to requests. Here is how you can make it happen. First, introduce a new kind of `CancellationTokenSource` for the delayed `CancellationToken` activation:
+The reason for this behavior is that ingress and Kubernetes control plane are separate entities, operating independently. When Kubernetes decides to shut down some pod(s), there is inevitable delay between control plane taking a pod out of Service backend, and ingress "noticing" that this has happened and actually stopping routing requests to that pod. So it is useful to introduce some delay between the time the pod receivers `SIGTERM` signal, and the time when it stops responding to requests. You can make it happen by replacing the default `IHostLifetime` implementation (which shuts down the application immediately upon receiving `SIGTERM` signal):
 
 ```csharp
-public class DelayedLinkedTokenSource
+using System.Runtime.InteropServices;
+
+public class DelayedShutdownHostLifetime : IHostLifetime, IDisposable
 {
-    private CancellationTokenSource _cts;
+    private IHostApplicationLifetime _applicationLifetime;
+    private TimeSpan _delay;
+    private IEnumerable<IDisposable> _disposables;
```

Comment:
It looks like nullability isn't enabled. That would be a good practice for our samples. For example, this member would be required to be nullable.

Reply:
Changed